pkg/rebuild/maven/maveninfer_test.go

```diff
@@ -0,0 +1,258 @@
+				},
+			},
+			targetPom: rebuild.Target{
+				Package: "com.example:child",
+				Version: "1.0.0",
+			},
+			expectedURL: "https://example.com/child",
+			wantErr:     false,
+		},
+		{
+			name: "get parent SCM URL",
+			pom: []PomXML{
+				{
+					GroupID:    "com.example",
+					ArtifactID: "child",
+					VersionID:  "1.0.0",
+					Parent: Parent{
+						GroupID:    "com.example.parent",
+						ArtifactID: "parent",
+						VersionID:  "1.0.0",
+					},
+				},
+				{
+					GroupID:    "com.example.parent",
+					ArtifactID: "parent",
+					VersionID:  "1.0.0",
+					SCMURL:     "https://github.com/example/parent",
+				},
+			},
+			targetPom: rebuild.Target{
+				Package: "com.example:child",
+				Version: "1.0.0",
+			},
+			expectedURL: "https://github.com/example/parent",
+			wantErr:     false,
+		},
+		{
+			name: "get URL of parent and not child",
+			pom: []PomXML{
+				{
+					GroupID:    "com.example",
+					ArtifactID: "child",
+					VersionID:  "1.0.0",
+					URL:        "https://example.com/child",
+					Parent: Parent{
+						GroupID:    "com.example.parent",
+						ArtifactID: "parent",
+						VersionID:  "1.0.0",
+					},
+				},
+				{
+					GroupID:    "com.example.parent",
+					ArtifactID: "parent",
+					VersionID:  "1.0.0",
+					URL:        "https://example.com/parent",
+				},
+			},
+			targetPom: rebuild.Target{
+				Package: "com.example:child",
+				Version: "1.0.0",
+			},
+			expectedURL: "https://example.com/parent",
+			wantErr:     false,
+		},
+		{
+			name: "should throw an error for no valid URL",
+			pom: []PomXML{
+				{
+					GroupID:    "com.example",
+					ArtifactID: "child",
+					VersionID:  "1.0.0",
+					Parent: Parent{
+						GroupID:    "com.example.parent",
+						ArtifactID: "parent",
+						VersionID:  "1.0.0",
+					},
+				},
+				{
+					GroupID:    "com.example.parent",
+					ArtifactID: "parent",
+					VersionID:  "1.0.0",
+				},
+			},
+			targetPom: rebuild.Target{
+				Package: "com.example:child",
+				Version: "1.0.0",
+			},
+			expectedURL: "",
+			wantErr:     true,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			mockRegistry := &mockMavenRegistry{
+				artifactCoordinates: make(map[struct{ PackageName, VersionID, FileType string }][]byte),
+			}
+			for _, pom := range tc.pom {
+				addPomArtifact(mockRegistry, &pom)
+			}
+			mockMux := rebuild.RegistryMux{
+				Maven: mockRegistry,
+			}
+			got, err := Rebuilder{}.InferRepo(context.Background(), tc.targetPom, mockMux)
+			if tc.wantErr {
+				if err == nil {
+					t.Fatalf("InferRepo() = %q, want error", got)
+				}
+			} else {
+				if err != nil {
+					t.Fatalf("InferRepo() error = %v", err)
+				}
+				if got != tc.expectedURL {
+					t.Errorf("InferRepo() = %q, want %q", got, tc.expectedURL)
+				}
+			}
+
+		})
+	}
+}
+
+func addPomArtifact(mavenRegistry *mockMavenRegistry, pom *PomXML) {
+	key := struct{ PackageName, VersionID, FileType string }{
```

Comment:
in a follow-on change, I think we should just name this type so we can stop using this verbose double-declare pattern